File: aop-tech/app/src/main/java/com/sososeen09/aop_tech/BindPhoneActivity.java
package com.sososeen09.aop_tech;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;
import com.sososeen09.aop_tech.aspect.TimeSpend;

public class BindPhoneActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_bind_phone);

        findViewById(R.id.btn_bind).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                attemptBind();
            }
        });
    }

    @TimeSpend("绑定手机号")
    private void attemptBind() {
        StatusHolder.sHasBindPhone = true;
        Toast.makeText(BindPhoneActivity.this, "绑定手机号成功", Toast.LENGTH_SHORT).show();
        finish();
    }
}

File: aop-tech/app-kotlin/src/main/java/com/sososeen09/aop/kotlin/BindPhoneActivity.kt
package com.sososeen09.aop.kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.Toast

class BindPhoneActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_bind_phone)

        findViewById<View>(R.id.btn_bind).setOnClickListener {
            StatusHolder.sHasBindPhone = true
            Toast.makeText(this@BindPhoneActivity, "绑定手机号成功", Toast.LENGTH_SHORT).show()
            finish()
        }
    }
}
